ci-operator/config/openshift/os/openshift-os-master.yaml

```diff
@@ -53,6 +64,64 @@ tests:
   container:
     from: build-test-qemu-img
   skip_if_only_changed: ^docs/|\.md$|^(?:.*/)?(?:\.gitignore|OWNERS|PROJECT|LICENSE)$
+- as: periodic-os-build
+  cluster: build02
+  cron: '@daily'
+  steps:
+    test:
+    - as: oc-login
+      cli: latest
+      commands: |
+        #!/bin/bash
+        set -euo pipefail
+        # To inject the oc CLI, ci-operator builds a new container from the
+        # provided container dependency (from:) and pushes it to the namespace
+        # Imagestream before creating a new pod to run the commands in. Because
+        # of the size of the coreos-assembler image, this adds quite some time
+        # to the process. So lets use a much smaller Fedora base image.
+        oc login https://api.ci.l2s4.p1.openshiftapps.com:6443 --token="$(cat /service-account-token/image-pusher-service-account-token)"
+        oc registry login --registry=registry.ci.openshift.org --to="$SHARED_DIR/dockercfg.json"
+      credentials:
+      - mount_path: /service-account-token
+        name: rhcos-devel-image-pusher-service-account-token
+        namespace: test-credentials
+      from: fedora
+      resources:
+        limits:
+          devices.kubevirt.io/kvm: "1"
+        requests:
+          cpu: 1000m
+          devices.kubevirt.io/kvm: "1"
+          memory: 3Gi
+    - as: build-test-and-push-img
+      commands: |
+        #!/bin/bash
+        set -euo pipefail
+
+        # Ensure we have a known directory for cosa to build in
+        export COSA_DIR="/tmp/cosa"
+        mkdir -p "$COSA_DIR"
+
+        # Run the cosa build / test
+        /src/ci/build-test-qemu.sh
+
+        export REGISTRY_AUTH_FILE="$SHARED_DIR/dockercfg.json"
+        # Ensure we're in the designated cosa directory so the push-container commands work
+        cd "$COSA_DIR"
+        # Tags with the cosa build ID / arch
+        cosa push-container registry.ci.openshift.org/rhcos-devel/rhel-coreos
+
+        cosa push-container registry.ci.openshift.org/rhcos-devel/rhel-coreos:latest
+        cosa push-container registry.ci.openshift.org/rhcos-devel/rhel-coreos:4.11
```

Comment:
So as far as I understand things, what happens at OCP product¹ branch time is that the `-master` presubmits are forked.  I *think* what might happen is that a crude substitution is made for e.g. `4.11` across the whole text to mean `4.10`?  Or perhaps it only replaces things that are *exactly* `4.11` inside a YAML value?  So what might make this work a bit better is if e.g. we injected `OCP_VERSION: "4.11"` as an environment variable block to the pod - that way the thing doing the branching doesn't need to be parsing shell script.

Related to this, we'll need to be careful about the `:latest` tag.  It may be better actually to have a separate distinct controller which mirrors `4.11` into `latest`.

I'm generally OK landing this as is, but on the flip side I think all of this will suddenly rear its head like a cobra popping up from the sand when 4.12 is a thing.

¹ It's confusing sometimes because we use OpenShift to run CI for OpenShift, so I try to use "OCP product" to clarify we're talking about how we develop/build OpenShift itself, not a build of something else inside OpenShift

Reply:
(Short term, it's probably OK for us to add a note to ourselves that when 4.11 branches to edit this file to drop the push to `:latest`)